Team8923_2021/src/main/java/org/firstinspires/ftc/team8923_2021/MasterAutonomous.java

```diff
@@ -0,0 +1,296 @@
+
+        //Set last know encoder values
+        lastEncoderRight = motorLeft.getCurrentPosition();
+
+        //set IMU heading offset
+        headingOffset = imu.getAngularOrientation().firstAngle - robotAngle;
+
+        telemetry.clear();
+        telemetry.update();
+        telemetry.addLine("Initialized. Ready to start!");
+    }
+
+    public void configureAutonomous(){
+        while(!isDoneSettingUp){
+            if(gamepad1.x){
+                alliance = Alliance.BLUE;
+            }else if(gamepad1.b){
+                alliance = Alliance.RED;
+            }
+            if(gamepad1.dpad_up){
+                delays++;
+            }else if(gamepad1.dpad_down){
+                delays--;
+            }
+            if(element == 0){
+                destination = Destinations.SQUAREA;
+
+            }else if(element == 1){
+                destination = Destinations.SQUAREB;
+            }else if(element == 4){
+                destination = Destinations.SQUAREC;
+            }
+
+            if(gamepad1.start){
+                isDoneSettingUp = true;
+            }
+            // input information
+            telemetry.addLine("Alliance Blue/Red: X/B");
+            telemetry.addLine("Add a delay: D-Pad Up/Down");
+            telemetry.addLine("toggle objective: a park/park and foundation");
+            telemetry.addLine("After routine is complete and robot is on field, press Start");
+
+            telemetry.addLine();
+            telemetry.addData("alliance: ", alliance);
+            telemetry.addData("delays:", delays);
+            telemetry.addData("destination", destination);
+            telemetry.update();
+        }
+    }
+
+    //using imu
+    public void imuPivot(double referenceAngle, double targetAngle, double maxSpeed, double kAngle, double timeout){
+        runtime.reset();
+        //counter-clockwise is positive
+        double pivot;
+        double currentRobotAngle;
+        double angleError;
+
+        targetAngle = referenceAngle + targetAngle;
+        targetAngle = adjustAngles(targetAngle);
+        do{
+            currentRobotAngle = imu.getAngularOrientation().firstAngle;
+            angleError =  currentRobotAngle - targetAngle;
+            angleError = adjustAngles(angleError);
+            pivot = angleError * kAngle;
+
+            if (pivot >= 0.0){
+                pivot = Range.clip(pivot, 0.15, maxSpeed);
+            }else{
+                pivot = Range.clip(pivot, -maxSpeed, -0.15);
+            }
+
+            speedLeft = pivot;
+            speedRight = -pivot;
+
+            motorLeft.setPower(speedLeft);
+            motorRight.setPower(speedRight);
+
+            idle();
+        }
+        while((opModeIsActive() && (Math.abs(angleError) > 3.0)) && (runtime.seconds() < timeout));
+        stopDriving();
+    }
+
+    public void reverseImuPivot(double referenceAngle, double targetAngle, double maxSpeed, double kAngle, double timeout){
```

Comment:
You can probably delete this method. The regular imuPivot method can go both directions if you pass a negative angle.

Reply:
I'm still seeing this method in the changes